### PR TITLE
Add user-agent header setting with tests

### PR DESCRIPTION
**Pull request by Altai-man**

### CODE REVIEW

File: lib/Cro/HTTP/Client.pm6 [hunk 1]
```diff
@@ -257,7 +258,7 @@ class Cro::HTTP::Client {
                     :$!body-serializers, :$!add-body-serializers,
                     :$!body-parsers, :$!add-body-parsers,
                     :$!follow, :%!auth, :$!http, :$!persistent = True, :$!ca,
-                    :$!push-promises) {
+                    :$!push-promises, :$!user-agent) {
```
Comment (jnthn) [inline lib/Cro/HTTP/Client.pm6]: This is a `BUILD`, so the default on the attribute will not take effect.

File: lib/Cro/HTTP/Client.pm6 [hunk 2]
```diff
@@ -539,6 +540,7 @@ class Cro::HTTP::Client {
         $request.append-header('Host', $url.host ~
             ($port && $port != 80 | 443 ?? ":$port" !! ""));
         if self {
+            $request.append-header('User-agent', $.user-agent) if $.user-agent && (%options<user-agent>:!exists);
```
Comment (jnthn) [inline lib/Cro/HTTP/Client.pm6]: This should only happen if nothing in `!set-headers` results in a `User-agent` header.

File: lib/Cro/HTTP/Client.pm6 [hunk 3]
```diff
@@ -587,6 +589,18 @@ class Cro::HTTP::Client {
...
             }
         }
+
+        # Set User-agent, check if wasn't set already for us
+        unless $request.has-header('User-agent') {
+            if %options<user-agent>:exists {
+                with %options<user-agent> {
+                    $request.append-header('User-agent', $_) if $_;
+                }
+            } else {
```
Comment (jnthn) [inline lib/Cro/HTTP/Client.pm6]: Probably we want an `elsif` here that is doing the instance-level check.

File: lib/Cro/HTTP/Client.pm6 (hunk 2 above)
Comment (Altai-man) [reply lib/Cro/HTTP/Client.pm6]: How does it change the logic?
If we work with an instance where user-agent was passed on instantiation and %options arg does not override it, we set the header, no?

**Comment by Altai-man** [comment]
@jnthn I am reviving this while you are on Cro spree...

**Comment by Altai-man** [comment]
It seems to fail tests related to json.

File: lib/Cro/HTTP/Client.pm6 (hunk 3 above)
Comment (jnthn) [reply lib/Cro/HTTP/Client.pm6]: This still applies; if you say at the client-level you want no user agent, then we should respect that.

**Pull request closed without merging**